src/forcing/ForcingsEngine.cpp

```diff
@@ -0,0 +1,239 @@
+#include "ForcingsEngine.hpp"
+
+namespace data_access {
+
+std::unordered_map<std::string, std::unique_ptr<ForcingsEngine>> ForcingsEngine::instances_{};
+
+ForcingsEngine::ForcingsEngine(const std::string& init, size_type time_start, size_type time_end)
+  : time_start_(std::chrono::seconds{time_start})
+  , time_end_(std::chrono::seconds{time_end})
+{
+    check_runtime_requirements();
+
+    bmi_ = std::make_unique<models::bmi::Bmi_Py_Adapter>(
+        "ForcingsEngine",
+        init,
+        "NextGen_Forcings_Engine.BMIForcingsEngine",
+        true,
+        true,
+        utils::getStdOut()
+    );
+    
+    time_step_ = std::chrono::seconds{static_cast<int64_t>(bmi_->GetTimeStep())};
+    var_outputs_ = bmi_->GetOutputVarNames();
+    var_outputs_.erase(var_outputs_.begin()); // Erase "CAT-ID" from outputs
```

Comment:
Maybe I'm being paranoid, but I don't feel like trusting the underlying code to always return `CAT-ID` as the *first* output variable. Could this be a `find` call instead of `begin`?